library/data/registry_entry_attachments/src/testmod/java/org/quiltmc/qsl/registry/attachment/test/DispatchedAttachmentTest.java

```diff
@@ -77,8 +79,20 @@ public TypedActionResult<ItemStack> use(World world, PlayerEntity user, Hand han
 	 */
 	public static final ModularFunctionItem ITEM_4 = Registry.register(Registry.ITEM,
 			new Identifier("quilt", "modular_item_4"), new ModularFunctionItem(new Item.Settings()));
+	/**
+	 * Has a value a provided by a tag.
+	 */
+	public static final ModularFunctionItem ITEM_5 = Registry.register(Registry.ITEM,
+			new Identifier("quilt", "modular_item_5"), new ModularFunctionItem(new Item.Settings()));
+
+	/**
+	 * Has a value a provided by a tag via datapack.
+	 */
+	public static final ModularFunctionItem ITEM_6 = Registry.register(Registry.ITEM,
+			new Identifier("quilt", "modular_item_6"), new ModularFunctionItem(new Item.Settings()));
 
 	@Override
 	public void onInitialize(ModContainer mod) {
+		RegistryEntryAttachmentHolder.getBuiltin(Registry.ITEM).putValue(MODULAR_FUNCTION, TagKey.of(Registry.ITEM_KEY, new Identifier("quilt", "modular_tag_1")), new SendMessageFuncValue("Provided via tag"));
```

Comment:
Setting a code-based value for a tag shouldn't require using internals, but I can't think of a good way to expose it (without scrapping the "built-in values can only be set during registration" idea, anyway)...

Reply:
Agree. Maybe i should expose this pattern a little clearer so that values can be set this way.